### ISSUE TITLE
Oversight on task_progress event?

### ISSUE THREAD

**Issue by sbasher314** [diamondio/better-queue#87]
Hello, I am recently working on a project using better-queue. I presume that the progressTask function should be able to work as a partial (say, inserted 10/100 records) and should be able to display that a task / ticket is 10% complete. However, utilizing either progressTask or progressBatch causes the ticket to stop being tracked / updated - I believe the below code is why:

```js
worker.on('task_progress', function (id, progress) {
    var taskId = taskIds[id];
    if (tickets[taskId]) {
      tickets[taskId].progress(progress);
      delete tickets[taskId];
    }
    self.emit('task_progress', taskId, progress);
  })
```

It looks like this is almost identical behavior to the worker.on('task_finish) listener, and may have erroneously been copied over:

```js
worker.on('task_finish', function (id, result) {
    var taskId = taskIds[id];
    var stats = updateStatsForEndedTask(taskId);
    if (tickets[taskId]) {
      tickets[taskId].finish(result);
      delete tickets[taskId];
    }
    self.emit('task_finish', taskId, result, stats);
  })
  ```

For my purposes, simply commenting out / removing the line `delete tickets[taskId]` in the task_progress listener above resolves this issue, and I do not see it breaking expected behavior. I just want to make sure that this is not the intended behavior, or if this conflicts with anything in any way.

**Comment by leanderlee** [comment]
I’m not sure I understand your issue completely. Can you provide some
example code to illustrate the problem?

**Comment by sbasher314** [comment]
Sure -

Lets say I have some number of operations to complete. We will use a for loop for simplicity - 

```js
for (let i = 1; i <= 100; i++) {
  this.progressTask(0, i, iterationCount, 'iterations')
}
``` 

After the first progressTask usage, the ticket is deleted from the worker / queue and does not receive further updates, such as the final result, status, or eta. The other events still mostly appear to work as expected.

**Comment by leanderlee** [comment]
Hmm, yeah, I think you're right. This does seem like an oversight... Weird that no one caught this! Fix incoming

**Comment by leanderlee** [comment]
Fix pushed to 3.8.11

**Comment by sbasher314** [comment]
Awesome, thanks!
